snapcraft/plugins/gulp.py

```diff
@@ -99,6 +99,14 @@ def build(self):
         env = os.environ.copy()
         env['PATH'] = '{}:{}'.format(
             os.path.join(self._npm_dir, 'bin'), env['PATH'])
+
+        # make npm use http proxy to access registry via https
+        # ref. https://github.com/npm/npm/issues/2050
+        if 'http_proxy' in os.environ:
```

Comment:
why don't you instead check for `https_proxy`? Or check for both and set both. What about `no_proxy`?